File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int cursum = 0;
        int maxi = Integer.MIN_VALUE;
        
        for(int i=0;i<nums.length;i++){
            cursum = cursum + nums[i];
            if(nums[i] > cursum){
                cursum = nums[i];
            }
            if(maxi < cursum){
                maxi = cursum;
            }
        }
        return maxi;
    }
}

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int cursum=0;
        int maxsum = nums[0];
        for(int i=0; i<nums.size(); i++){
            cursum += nums[i];
            if(nums[i] > cursum){
                cursum = nums[i];
            }
            if(maxsum < cursum){
                maxsum = cursum;
            }
        }
        return maxsum;
    }
};